lms/djangoapps/instructor/views/tools.py

```diff
@@ -320,3 +320,36 @@ def add_block_ids(payload):
         for ele in payload['data']:
             if 'module_id' in ele:
                 ele['block_id'] = UsageKey.from_string(ele['module_id']).block_id
+
+
+def generate_course_forums_d3(url_handle):
+    """
+    Grab the csv file at url_handle and parse it. Return a new-line joined csv string to be graphed
+    """
+    reader = csv.reader(url_handle)
+    reader.next()
+    previous_date = -1
+    previous = None
+    rows = [['Date', 'New Threads', 'Responses', 'Comments']]
+    for date, type, number, up, down, net in reader:
```

Comment:
There's a couple of pylint violations for this line:

334: C0103: (invalid-name), generate_course_forums_d3: Invalid name "up" for type variable (should match [a-z_][a-z0-9_]{2,30}$)
    334: W0622: (redefined-builtin), generate_course_forums_d3: Redefining built-in 'type'